components/camel-sjms2/src/test/java/org/apache/camel/component/sjms2/consumer/InOnlyTopicDurableConsumerTest.java

```diff
@@ -49,9 +53,8 @@ public void testDurableTopic() throws Exception {
         mock2.expectedBodiesReceived("Hello World");
 
         // wait a bit and send the message
-        Thread.sleep(1000);
-
-        template.sendBody("sjms2:topic:foo", "Hello World");
+        Awaitility.await().pollDelay(1, TimeUnit.SECONDS).untilAsserted(() -> Assertions
```

Comment:
This would not really work correct as you risk sending more than 1 message if this code is executed again by awaitability.

So you cannot do this when you do `template.sendBody...` as that code is supposed to only be executed once.

Reply:
Oh okay, thanks!
I will see if there is any other way to replace thread.sleep.

Reply:
@davsclaus i have updated the code. Can you please review it?
Thanks!